README.md

```diff
@@ -7,3 +7,216 @@ Status](https://github.com/maltezfaria/DataFlowTasks.jl/workflows/CI/badge.svg)]
+- `1` cholesky factorization of the (i,i) block (I),
+- `(i-1)` triangular solves (one for each block in the `i`-th row) (II),
+- `i*(i-1)/2` matrix multiplications to update the submatrix (III).
+
+We illustrate the 2nd step of the algorithm in the following image.
+
+![Cholesky_Image](docs/src/Cholesky_2ndStep.png)
+
+A sequential tiled factorization algorithm can be implemented as:
+
+```julia
+tilerange(ti, ts) = (ti-1)*ts+1:ti*ts
+function cholesky_tiled!(A, ts)
+    m,n = size(A)
+    n%ts != 0 && error("Tilesize doesn't fit the matrix")
+    tn = n÷ts
+
+    for ti in 1:tn
+        ri = tilerange(ti, ts)
+
+        # Diagonal cholesky serial factorization (I)
+        cholesky!(view(A,ri,ri))
+
+        # Left blocks update (II)
+        L = adjoint(UpperTriangular(view(A,ri,ri)))
+        for tj in ti+1:tn
+            rj = tilerange(tj, ts)
+            ldiv!(L, view(A,ri,rj))
+        end
+
+        # Submatrix update (III)
+        for tj in ti+1:tn
+            for tk in tj:tn
+                rj = tilerange(tj, ts)  ;  rk = tilerange(tk, ts)
+                mul!(view(A,rj,rk), adjoint(view(A,ri,rj)), view(A,ri,rk))
+            end
+        end
+    end
+
+    # Construct the factorized object
+    return Cholesky(A,'U',zero(LinearAlgebra.BlasInt))
+end
+```
+
+In order to parallelize the code with DataFlowTasks, it will only be necessary to wrap function calls within a `@dspawn`, and add a synchronization point at the end. The parallelized code will be :
+
+```julia
+using DataFlowTasks
+using LinearAlgebra
+function cholesky_tiled!(A, ts)
+    m,n = size(A)
+    n%ts != 0 && error("Tilesize doesn't fit the matrix")
+    tn = n÷ts
+
+    Atile = [view(A, tilerange(ti, ts), tilerange(tj, ts)) for ti in 1:tn, tj in 1:tn]
+
+    for ti in 1:tn
+        # Diagonal cholesky serial factorization (I)
+        @dspawn cholesky!(@RW(Atile[ti,ti])) label="chol ($ti,$ti)"
+
+
+        # Left blocks update (II)
+        L = adjoint(UpperTriangular(Atile[ti,ti]))
+        for tj in ti+1:tn
+            @dspawn ldiv!(@R(L), @RW(Atile[ti,tj])) label="ldiv ($ti,$tj)"
+        end
+
+        # Submatrix update (III)
+        for tj in ti+1:tn
+            for tk in tj:tn
+                Ajk = Atile[tj,tk]
+                Aik = Atile[ti,tk]
+                Aji = adjoint(Atile[ti,tj])
+                @dspawn mul!(@RW(Ajk), @R(Aji), @R(Aik), -1, 1) label="schur ($tj,$tk)"
+            end
+        end
+    end
+    # Construct the factorized object
+    r = @dspawn Cholesky(A,'U',zero(LinearAlgebra.BlasInt))
+    return fetch(r)
+end
+```
+
+If you run this code, you'll see `memory_overlap` warning on the fact that there's no method to evaluate the memory overlaping between an `Adjoint` and a `SubArray`. So we have to implement this method :
+
+```julia
+using DataFlowTasks: memory_overlap
+memory_overlap(U::Adjoint,A) = memory_overlap(U.parent,A)
+memory_overlap(U,L::Adjoint) = memory_overlap(L,U)
+```
```

Comment:
Please don't advise users to do this: it is type piracy

Well, technically it isn't here because this code defines a new `memory_overlap` function instead of adding methods to `DataFlowTasks.memory_overlap`, but this version shouldn't work either...